dashboard/app/views/devise/registrations/new.html.haml

```diff
@@ -94,23 +114,131 @@
     %br/
 
 :javascript
+
+  $(document).ready(function() {
+    // Check for URL having: /users/sign_up?user%5Buser_type%5D=teacher
+    if ("#{resource.teacher?}" == "true") {
+      showTeacher();
+    }
+  });
+
+  function formSuccess() {
+    console.log('form success');
+    window.location.href = '/';
+  }
+
+  function formError(err) {
+    //console.log('form error', err);
+
+    // Define the fields that can have specific errors attached to them.
+    var fields = ["name", "email", "password", "password_confirmation", "schoolname", "age", "gender"];
+
+    for (var i = 0; i < fields.length; i++) {
+      var field = fields[i];
+      if (err.responseJSON.errors[field]) {
+        var errorField = $("#" + field + "-block .error_in_field");
+        errorField.text(err.responseJSON.errors[field]);
+        errorField.fadeTo("normal", 1);
+      }
+    }
+  }
+
+  function changeRole(value) {
+    if (value == "student") {
+      showStudent();
+    } else if (value == "teacher") {
+      showTeacher();
+    }
+  }
+
+  function showStudent() {
+    // show age (and reset to blank)
+    $("#age-block").fadeIn();
+
+    // show gender-optional
```

Comment:
Not sure these comments are helpful (except for schoolcountry)? Also if age can be reset to blank and the other comments don't mention that, does that mean gender isn't reset? Why only age?